src/Mutator/MutatorFactory.php

```diff
@@ -82,33 +76,16 @@ public function create(array $resolvedMutators): array
             /** @var string[] $ignored */
             $ignored = $config['ignore'] ?? [];
 
-            switch ($mutatorClass) {
-                case BCMath::class:
-                    $mutator = new BCMath(new BCMathConfig($settings));
+            if (in_array(ConfigurableMutator::class, class_implements($mutatorClassName), true)) {
+                $configClassName = $mutatorClassName::getConfigClassName();
 
-                    break;
-
-                case MBString::class:
-                    $mutator = new MBString(new MBStringConfig($settings));
-
-                    break;
-
-                case TrueValue::class:
-                    $mutator = new TrueValue(new TrueValueConfig($settings));
-
-                    break;
-
-                case ArrayItemRemoval::class:
-                    $mutator = new ArrayItemRemoval(new ArrayItemRemovalConfig($settings));
-
-                    break;
-
-                default:
-                    /** @var Mutator $mutator */
-                    $mutator = new $mutatorClass();
+                $mutator = new $mutatorClassName(new $configClassName($settings));
+            } else {
+                $mutator = new $mutatorClassName();
             }
+            /* @var Mutator $mutator */
 
-            $mutators[$mutator->getName()] = new IgnoreMutator(
+            $mutators[(string) $mutator->getName()] = new IgnoreMutator(
```

Comment:
`public function getName(): string;`

why is it needed?

Reply:
I think it was an SA error, will double check

Reply:
```
Method Infection\Mutator\MutatorFactory::create() should return array<string, Infection\Mutator\Mutator> but returns array<int|string,
         Infection\Mutator\IgnoreMutator>.
```

It is unclear to me why...